go/vt/topo/etcd2topo/server_test.go

```diff
@@ -89,6 +95,122 @@ func startEtcd(t *testing.T) (*exec.Cmd, string, string) {
 	return cmd, dataDir, clientAddr
 }
 
+// startEtcdWithTLS starts an etcd subprocess with TLS setup, and waits for it to be ready.
+func startEtcdWithTLS(t *testing.T) (*exec.Cmd, string, string) {
+	// Create a temporary directory.
+	dataDir, err := ioutil.TempDir("", "etcd")
+	if err != nil {
+		t.Fatalf("cannot create tempdir: %v", err)
+	}
+
+	// Get our two ports to listen to.
+	port := testfiles.GoVtTopoEtcd2topoPort
+	name := "vitess_unit_test"
+	clientAddr := fmt.Sprintf("http://localhost:%v", port)
+	peerAddr := fmt.Sprintf("http://localhost:%v", port+1)
+	initialCluster := fmt.Sprintf("%v=%v", name, peerAddr)
+
+	// Generate cert and key in /tmp
+	err = exec.Command("openssl", "req", "-new", "-newkey", "rsa:4096", "-days", "365", "-nodes", "-x509",
+		"-subj", "/C=US/ST=Denial/L=Springfield/O=Dis/CN=www.example.com",
+		"-keyout", testKeyPath, "-out", testCertPath).Run()
+	if err != nil {
+		t.Fatalf("unable to generate test key and cert using openssl: %v", err)
+	}
+
+	cmd := exec.Command("etcd",
+		"-name", name,
+		"-advertise-client-urls", clientAddr,
+		"-initial-advertise-peer-urls", peerAddr,
+		"-listen-client-urls", clientAddr,
+		"-listen-peer-urls", peerAddr,
+		"-initial-cluster", initialCluster,
+		"-cert-file", testCertPath,
+		"-key-file", testKeyPath,
+		"-data-dir", dataDir)
+	err = cmd.Start()
+	if err != nil {
+		t.Fatalf("failed to start etcd: %v", err)
+	}
+
+	// Safe now to build up TLS info.
+	tlsInfo := transport.TLSInfo{
+		CertFile:   testCertPath,
+		KeyFile:    testKeyPath,
+		TrustedCAFile: *caPath,
+	}
+
+	tlsConfig, err := tlsInfo.ClientConfig()
+
+	// Create a client to connect to the created etcd.
+	cli, err := clientv3.New(clientv3.Config{
+		Endpoints:   []string{clientAddr},
+		TLS: tlsConfig,
+		DialTimeout: 5 * time.Second,
+	})
+	if err != nil {
+		t.Fatalf("newCellClient(%v) failed: %v", clientAddr, err)
+	}
```

Comment:
We should probably `defer cli.Close()` even though this is just a test.

Reply:
Done.